File: Planarian.Web/src/Modules/Account/Models/InvitationEmailHistoryVm.ts
import { MessageDeliveryEventType } from "../../../Shared/Models/MessageDeliveryEventType";
import { MessageDeliveryStatus } from "../../../Shared/Models/MessageDeliveryStatus";

export interface InvitationEmailEventVm {
  eventType: MessageDeliveryEventType;
  occurredOn: string;
  bot?: string | null;
  severity?: string | null;
  reason?: string | null;
  deliveryCode?: string | null;
  enhancedDeliveryCode?: string | null;
  attemptNumber?: number | null;
  isDelayedBounce?: boolean | null;
}

export interface InvitationEmailAttemptVm {
  messageLogId: string;
  createdOn: string;
  deliveryStatus?: MessageDeliveryStatus | null;
  deliveryStatusOn?: string | null;
  events: InvitationEmailEventVm[];
}

File: Planarian/Planarian/Modules/Users/Models/InvitationEmailHistoryVm.cs
using Planarian.Model.Shared;

namespace Planarian.Modules.Users.Models;

public class InvitationEmailAttemptVm
{
    public string MessageLogId { get; set; } = null!;
    public DateTime CreatedOn { get; set; }
    public MessageDeliveryStatus? DeliveryStatus { get; set; }
    public DateTime? DeliveryStatusOn { get; set; }
    public List<InvitationEmailEventVm> Events { get; set; } = new();
}

public class InvitationEmailEventVm
{
    public MessageDeliveryEventType EventType { get; set; }
    public DateTime OccurredOn { get; set; }
    public string? Bot { get; set; }
    public string? Severity { get; set; }
    public string? Reason { get; set; }
    public string? DeliveryCode { get; set; }
    public string? EnhancedDeliveryCode { get; set; }
    public int? AttemptNumber { get; set; }
    public bool? IsDelayedBounce { get; set; }
}
